In [656]:
import pandas as pd
import warnings
import seaborn as sns
import numpy as np
warnings.filterwarnings("ignore")

In [657]:
df = pd.read_csv('liste-des-musees-de-france-2021.csv',error_bad_lines=False,encoding='latin-1',sep=';')
df.head()

osm_id                         name number                      street  \
0  4540234591                      3 bis f    109  Avenue du Petit Barthélémy   
1  4388745689             A casa di Losari    NaN                        T 30   
2  4426176194             À l'homme volant     35                  Grande Rue   
3  7652598615  Abbadiale - Maison des arts    NaN              Rue du Gabizos   
4   121580330                       Abbaye    NaN             Rue de l'Abbaye   

   postal_code              city country country_code        lat       lon  \
0      13617.0   Aix-en-Provence  France           fr  43.521207  5.439875   
1      20226.0         Belgodère  France           fr  42.638732  9.007660   
2      72300.0  Sablé-sur-Sarthe  France           fr  47.839656 -0.334043   
3      65400.0  Arras-en-Lavedan  France           fr  42.990826 -0.128057   
4      39210.0    Château-Chalon  France           fr  46.753866  5.624296   

                     website              email        phone  fax  \
0       http://www.3bisf.com  contact@3bisf.com  33442161775  NaN   
1                        NaN                NaN          NaN  NaN   
2                        NaN                NaN  33243950736  NaN   
3  https://www.abbadiale.fr/                NaN  33633170714  NaN   
4                        NaN                NaN          NaN  NaN   

                                      tags description   date_added wikidata  
0          osm:museum;art:art contemporain         NaN  44116,00764      NaN  
1                               osm:museum         NaN   43338,9394      NaN  
2  osm:museum;type:centre d'interpretation         NaN  42644,46366      NaN  
3                               osm:museum         NaN  44120,04668      NaN  
4              osm:museum;art:architecture         NaN  43102,54919      NaN

In [658]:
# Rows and Columns
df.shape

(3784, 18)

In [659]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3784 entries, 0 to 3783
Data columns (total 18 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   osm_id        3784 non-null   int64  
 1   name          3784 non-null   object 
 2   number        743 non-null    object 
 3   street        3771 non-null   object 
 4   postal_code   3728 non-null   float64
 5   city          3782 non-null   object 
 6   country       3784 non-null   object 
 7   country_code  3784 non-null   object 
 8   lat           3784 non-null   float64
 9   lon           3784 non-null   float64
 10  website       1636 non-null   object 
 11  email         301 non-null    object 
 12  phone         1211 non-null   object 
 13  fax           0 non-null      float64
 14  tags          3782 non-null   object 
 15  description   226 non-null    object 
 16  date_added    3775 non-null   object 
 17  wikidata      1147 non-null   object 
dtypes: float64(4), int64(1), obj

In [660]:
#Basics stats

In [661]:
df.describe(include="all")

osm_id                  name number      street    postal_code  \
count   3.784000e+03                  3784    743        3771    3728.000000   
unique           NaN                  3466    196        2932            NaN   
top              NaN  Musée des Beaux-Arts      1  Grande Rue            NaN   
freq             NaN                    23     73          34            NaN   
mean    2.193306e+09                   NaN    NaN         NaN   48499.273069   
std     2.521168e+09                   NaN    NaN         NaN   27982.341434   
min     5.107373e+06                   NaN    NaN         NaN    1000.000000   
25%     1.451896e+08                   NaN    NaN         NaN   25582.500000   
50%     8.226927e+08                   NaN    NaN         NaN   49730.000000   
75%     3.982632e+09                   NaN    NaN         NaN   71570.000000   
max     8.394269e+09                   NaN    NaN         NaN  590007.000000   

         city country country_code          lat          lon  \
count    3782    3784         3784  3784.000000  3784.000000   
unique   2295       2            2          NaN          NaN   
top     Paris  France           fr          NaN          NaN   
freq      175    3776         3776          NaN          NaN   
mean      NaN     NaN          NaN    46.531075     2.724237   
std       NaN     NaN          NaN     2.234572     2.897485   
min       NaN     NaN          NaN    41.498794    -5.129406   
25%       NaN     NaN          NaN    44.542494     0.570318   
50%       NaN     NaN          NaN    46.513252     2.563655   
75%       NaN     NaN          NaN    48.639156     4.984839   
max       NaN     NaN          NaN    51.053756     9.488302   

                            website                   email        phone  fax  \
count                          1636                     301         1211  0.0   
unique                         1585                     295         1163  NaN   
top     http://www.ville-orleans.fr  musees@ville.angers.fr  33325868686  NaN   
freq                              4                       3            4  NaN   
mean                            NaN                     NaN          NaN  NaN   
std                             NaN                     NaN          NaN  NaN   
min                             NaN                     NaN          NaN  NaN   
25%                             NaN                     NaN          NaN  NaN   
50%                             NaN                     NaN          NaN  NaN   
75%                             NaN                     NaN          NaN  NaN   
max                             NaN                     NaN          NaN  NaN   

              tags description  date_added  wikidata  
count         3782         226        3775      1147  
unique         119         206        3331      1133  
top     osm:museum    Ecomusée  43940,5849  Q3329951  
freq          2175          19          76         2  
mean           NaN         NaN         NaN       NaN  
std            NaN         NaN         NaN       NaN  
min            NaN         NaN         NaN       NaN  
25%            NaN         NaN         NaN       NaN  
50%            NaN         NaN         NaN       NaN  
75%            NaN         NaN         NaN       NaN  
max            NaN         NaN         NaN       NaN

In [662]:
# Basics stats for the object variables
df.describe(include=object)

name number      street   city country country_code  \
count                   3784    743        3771   3782    3784         3784   
unique                  3466    196        2932   2295       2            2   
top     Musée des Beaux-Arts      1  Grande Rue  Paris  France           fr   
freq                      23     73          34    175    3776         3776   

                            website                   email        phone  \
count                          1636                     301         1211   
unique                         1585                     295         1163   
top     http://www.ville-orleans.fr  musees@ville.angers.fr  33325868686   
freq                              4                       3            4   

              tags description  date_added  wikidata  
count         3782         226        3775      1147  
unique         119         206        3331      1133  
top     osm:museum    Ecomusée  43940,5849  Q3329951  
freq          2175          19          76         2

#### 1.-Combien y-a-t-il de musées en France métropolitaine ?

In [663]:
df["name"].value_counts().sum()

3784

In [664]:
df["website"].value_counts().sum()

1636

In [665]:
len(df["name"])

3784

* La France métropolitaine compte 3784 musées.

#### 2.-Dans quelle(s) ville(s) y-a-t-il de plus de musées ?

In [666]:
museums_by_city = df.city.value_counts()[0:50]
print("Les dix villes qui ont plus de musées\n{}".format(museums_by_city))

Les dix villes qui ont plus de musées
Paris                    175
Lyon                      33
Marseille                 28
Grenoble                  22
Bordeaux                  20
Toulouse                  20
Nice                      16
Strasbourg                16
Rouen                     15
Lille                     15
Reims                     13
Angers                    13
Montpellier               13
Aix-en-Provence           13
Avignon                   11
Dijon                     11
Colmar                    11
Perpignan                 10
Tours                     10
Besançon                  10
Digne-les-Bains            9
Nîmes                      9
Auxerre                    9
Saint-Étienne              9
Nantes                     9
Rochefort                  9
Orléans                    9
Rodez                      8
Toulon                     8
Mulhouse                   8
Limoges                    8
Monaco                     8
Thiers                     8
Cherb

#### 3.-Quel est le nombre moyen de musées par ville ?

In [667]:
meanMuseums_by_city = df.city.value_counts().mean()
print("Le nombre moyen de musée par ville est {}".format(meanMuseums_by_city))

Le nombre moyen de musée par ville est 1.647930283224401


#### 4.-Quel est le nombre médian de musées par ville ?

In [686]:
medianMuseum_by_city = museums_by_city.median()
print(medianMuseum_by_city)

9.0


#### 5.-Comment sont répartis les musées par type (en pourcentage) ?

In [689]:
display(df['tags'].value_counts(normalize=True)[0:50])

osm:museum                                                                        0.575093
osm:museum;type:ecomusee                                                          0.102327
osm:museum;type:musee technique et industriel                                     0.051296
osm:museum;type:musee d'histoire                                                  0.034902
osm:museum;art:architecture                                                       0.031994
osm:museum;art:prehistoire;type:musee archeologique                               0.022475
osm:museum;type:musee de beaux-arts                                               0.018509
osm:museum;type:musee d'art                                                       0.015071
osm:museum;type:musee d'histoire naturelle                                        0.014543
osm:museum;type:musee d'arts populaires                                           0.012163
osm:museum;type:centre d'interpretation                                           0.010841

#### 6.-Combien y-a-t-il de musées dont le nom commence par "Château" ?

In [691]:
ds = df["name"] == "Château"
ds.value_counts()

False    3783
True        1
Name: name, dtype: int64

In [692]:
# Deuxième méthode
def nombre_occurence(L,a):
    i=0
    for x in L:
        if (x== "Château"):
            i =i+1
    return i

In [693]:
print("Le nombre d'occurence de Château est :",nombre_occurence(name,"Château"))

Le nombre d'occurence de Château est : 1


#### 7.- Pour combien de musées dispose-t-on de l'adresse du site web ?

In [694]:
df.columns

Index(['osm_id', 'name', 'number', 'street', 'postal_code', 'city', 'country',
       'country_code', 'lat', 'lon', 'website', 'email', 'phone', 'fax',
       'tags', 'description', 'date_added', 'wikidata'],
      dtype='object')

In [695]:
Museum_with_website = df['website'].value_counts().sum()
print(" Le nombre de musées qui disposent de website est de {}".format(Museum_with_website))

 Le nombre de musées qui disposent de website est de 1636


#### 8.-Quel département français possède le plus de musées sur son territoire ?

In [696]:
most_museums_by_dep = df.postal_code.value_counts().idxmax()
print(int(most_museums_by_dep))

75005


#### 9.- Quel département français possède le moins de musées sur son territoire ?

In [697]:
less_museums_by_dep = df.postal_code.value_counts().idxmin()
print(int(less_museums_by_dep))

56630


#### 10.-Combien de musées ont "Napoléon" dans leur nom ?

In [698]:
def nombre_occurence(L,a):
    i=0
    for x in L:
        if (x== "Napoléon"):
            i =i+1
    return i

In [699]:
print("Le nombre d'occurence de Napoléon est :",nombre_occurence(name,"Napoléon"))

Le nombre d'occurence de Napoléon est : 0


In [701]:
# 2e méthode
ds = df["name"] == "Napoléon"
ds.value_counts()

False    3784
Name: name, dtype: int64